stratum/hal/bin/barefoot/main.cc

```diff
@@ -152,11 +153,12 @@ Main(int argc, char* argv[]) {
   std::unique_ptr<DeviceMgr> device_mgr(new DeviceMgr(unit));
 
   auto pi_node = pi::PINode::CreateInstance(device_mgr.get(), unit);
+  auto onlp_wrapper = OnlpWrapper::Make().ConsumeValueOrDie();
```

Comment:
I think OnlpWrapper should be a singleton (`CreateInstance`).

`onlp_sw_init` should probably not called multiple times.